base/compiler/optimize.jl

```diff
@@ -828,9 +832,9 @@ function inline_worthy(ir::IRCode,
         stmt = ir.stmts[line][:inst]
         thiscost = statement_or_branch_cost(stmt, line, ir, ir.sptypes, union_penalties, params)
         bodycost = plus_saturate(bodycost, thiscost)
-        bodycost > cost_threshold && return false
+        bodycost > cost_threshold && return MAX_INLINE_COST
```

Comment:
In this PR, or maybe better to be a followup, we probably want to change this function to `inline_cost`, and have it return `clamp(div(bodycost, 10), UInt16)`, and then the caller decides `inline_worthy` based on its own thresholds

Reply:
👍 I will have a followup PR to this with the changes that Shuhei mentioned as well